pallets/file-system/src/utils.rs

```diff
@@ -436,18 +434,17 @@ where
             Error::<T>::BspAlreadyVolunteered
         );
 
-        // Compute BSP's xor
-        let bsp_xor: T::ThresholdType = Self::compute_bsp_xor(
-            storage_request_metadata
-                .fingerprint
-                .as_ref()
-                .try_into()
-                .map_err(|_| Error::<T>::FailedToEncodeFingerprint)?,
-            &bsp_id
-                .encode()
-                .try_into()
-                .map_err(|_| Error::<T>::FailedToEncodeBsp)?,
-        )?;
+        // Concatenate BSP ID with the file's fingerprint and hash the concatenation to get the volunteering hash.
+        let concatenated = sp_std::vec![
+            bsp_id.encode(),
+            storage_request_metadata.fingerprint.encode()
+        ]
+        .concat();
+        let volunteering_hash =
+            <<T as frame_system::Config>::Hashing as Hash>::hash(concatenated.as_ref());
```

Comment:
Let's abstract this in a function cause we will probably do a runtime API for it at some point, to have a unique place where this calculation is coded and so that it can be called from the BSP Client.